Documentation/botr/ilc-architecture.md

```diff
@@ -0,0 +1,186 @@
+# ILC Compiler Architecture
+
+Author: Michal Strehovsky ([@MichalStrehovsky](https://github.com/MichalStrehovsky)) - 2018
+
+ILC (IL Compiler) is an ahead of time compiler that transforms programs in CIL (Common Intermediate Language) into a target language or instruction set to be executed on the CoreRT runtime (or CoreCLR runtime, in the future). The input to ILC is the common instruction format generated by popular managed language compilers such as C#, VB.NET, or F#. The output of ILC is native code for the target platform, along with data structures required to support executing the code on the target runtime. With a bit of stretch, one could say that ILC is an ahead of time compiler for C#.
```

Comment:
> one could say that ILC is an ahead of time compiler for C#.

Did you perhaps mean to say "native compiler for C#"? The paragraph opens with "is an ahead of time compiler" so it's not clear what value the "for C#" qualification adds here.